### ISSUE TITLE
navbar login state desync and invlid state, results in visual bug

### ISSUE THREAD

**Issue by T3sT3ro** [modrinth/code#1489]
### Describe the bug

Follow this thread for more info: https://discord.com/channels/734077874708938864/1153300605826248765

TL;DR is that the navbar enters some invalid state under certain conditions, where it displays controls for logged out users and controls for logged in users.

This in turn leads to a visual bug like this one:
![image](https://github.com/modrinth/knossos/assets/5300963/c18eb76c-016d-4eeb-8965-2b2d57c55d8c)

### Steps to reproduce

1. make sure you are not logged in on modrinth
2. search for mod in google
3. enter mod page, you should be logged out
4. login via github (other methods not tested)
5. go back in history to be on the same page as in step 3. (back button in browser x2 times in my case)
6. The default, non logged out navbar version is presented, after a short while the logged in status is added in, breaking the UI.

### Expected behavior

The navbar should show appropriate state at all times, possibly swapping the logged out to logged in version when it updates instead of updating it.

### Additional context

Chrome browser, Github login.

**Comment by Minenash** [comment]
Well, it no longer causes the both issue anymore, but it does cause a different issue, #2266
